pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/SegmentDeletionManager.java

```diff
@@ -165,40 +167,55 @@ protected synchronized void deleteSegmentFromPropertyStoreAndLocal(String tableN
 
   public void removeSegmentsFromStore(String tableNameWithType, List<String> segments) {
     for (String segment : segments) {
-      removeSegmentFromStore(tableNameWithType, segment);
+      removeSegmentFromStore(tableNameWithType, segment, _controllerDefaultDeletedSegmentsRetentionInDays);
     }
   }
 
-  protected void removeSegmentFromStore(String tableNameWithType, String segmentId) {
+  protected void removeSegmentFromStore(String tableNameWithType, String segmentId,
+      int deletedSegmentsRetentionInDays) {
```

Comment:
Remove this new argument, and use `_controllerDefaultDeletedSegmentsRetentionInDays` instead